src/workerd/io/BUILD.bazel

```diff
@@ -154,35 +157,55 @@ wd_cc_library(
     ],
 )
 
-# TODO(cleanup): Split this up further.
 wd_cc_library(
-    name = "actor",
-    srcs = [
-        "actor-cache.c++",
-        "actor-sqlite.c++",
-        "actor-storage.c++",
-    ],
-    hdrs = [
-        "actor-cache.h",
-        "actor-sqlite.h",
-        "actor-storage.h",
-    ],
+    name = "actor-cache",
+    srcs = ["actor-cache.c++"],
+    hdrs = ["actor-cache.h"],
     implementation_deps = [
-        "@sqlite3",
+        ":actor-storage",
+        ":io-gate",
+        "//src/workerd/util:duration-exceeded-logger",
+        "//src/workerd/util:sentry",
     ],
     visibility = ["//visibility:public"],
     deps = [
         ":actor-storage_capnp",
+        "//src/workerd/jsg:exception",
+        "@capnp-cpp//src/capnp:capnp-rpc",
```

Comment:
nit: capnp-rpc should not be needed here, we don't include any capnp headers in actor-cache.h